src/components/chat-messages.tsx

```diff
@@ -0,0 +1,202 @@
+import stripAnsi from 'strip-ansi';
+import { useExecuteWPCLI } from '../hooks/use-execute-cli';
+import { cx } from '../lib/cx';
+import { getIpcApi } from '../lib/get-ipc-api';
+import Button from './button';
+import { ExecuteIcon } from './icons/execute';
+
+interface MessageProps {
+	children: React.ReactNode;
+	isUser: boolean;
+	id?: number;
+	className?: string;
+	projectPath?: string;
+	siteId?: string;
+	blocks?: {
+		cliOutput?: string;
+		cliStatus?: 'success' | 'error';
+		cliTime?: string;
+		codeBlockContent?: string;
+	}[];
+	updateMessage?: (
+		id: number,
+		content: string,
+		output: string,
+		status: 'success' | 'error',
+		time: string
+	) => void;
+}
+
+interface InlineCLIProps {
+	output?: string;
+	status?: 'success' | 'error';
+	time?: string | null;
+}
+
+const InlineCLI = ( { output, status, time }: InlineCLIProps ) => (
+	<div className="p-3 bg-[#2D3337]">
+		<div className="flex justify-between mb-2 font-sans">
+			<span className={ status === 'success' ? 'text-[#63CE68]' : 'text-[#E66D6C]' }>
+				{ status === 'success' ? __( 'Success' ) : __( 'Error' ) }
+			</span>
+			<span className="text-gray-400">{ time }</span>
+		</div>
+		<pre className="text-white !bg-transparent !m-0 !px-0">
+			<code className="!bg-transparent !mx-0 !px-0">{ output }</code>
+		</pre>
+	</div>
+);
+
+const ActionButton = ( {
+	primaryLabel,
+	secondaryLabel,
+	icon,
+	onClick,
+	timeout,
+	disabled,
+}: {
+	primaryLabel: string;
+	secondaryLabel: string;
+	icon: JSX.Element;
+	onClick: () => void;
+	timeout?: number;
+	disabled?: boolean;
+} ) => {
+	const [ buttonLabel, setButtonLabel ] = useState( primaryLabel );
+
+	const handleClick = () => {
+		onClick();
+		setButtonLabel( secondaryLabel );
+		if ( timeout ) {
+			setTimeout( () => {
+				setButtonLabel( primaryLabel );
+			}, timeout );
+		}
+	};
+
+	return (
+		<Button
+			onClick={ handleClick }
+			variant="tertiary"
+			className="mr-2 font-sans select-none"
+			disabled={ disabled }
+		>
+			{ icon }
+			<span className="ml-1">{ buttonLabel }</span>
+		</Button>
+	);
+};
+
+export const Message = ( {
+	children,
+	id,
+	isUser,
+	className,
+	projectPath,
+	blocks,
+	updateMessage,
+}: MessageProps ) => {
+	const CodeBlock = ( props: JSX.IntrinsicElements[ 'code' ] & ExtraProps ) => {
+		const content = String( props.children ).trim();
+		const containsWPCommand = /\bwp\s/.test( content );
+		const wpCommandCount = ( content.match( /\bwp\s/g ) || [] ).length;
+		const containsSingleWPCommand = wpCommandCount === 1;
+
+		const {
+			cliOutput,
+			cliStatus,
+			cliTime,
+			isRunning,
+			handleExecute,
+			setCliOutput,
+			setCliStatus,
+			setCliTime,
+		} = useExecuteWPCLI( content, projectPath, updateMessage, id );
+
+		useEffect( () => {
+			if ( blocks ) {
+				const block = blocks?.find( ( block ) => block.codeBlockContent === content );
+				if ( block ) {
+					setCliOutput( block?.cliOutput ? stripAnsi( block.cliOutput ) : null );
```

Comment:
[Suggestion] I suggest to change that to `setBlock`, as this is what those variables actually represent